# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1450175425 -39600
#      Tue Dec 15 21:30:25 2015 +1100
# Node ID 87b3348d8d76a4cbf63d2829750d3ebeb998756f
# Parent  a1529d24beff0bf545fe74f54341dd713c717ff7
Fix clearing breakpoints in @class files with dbstatus (bug #41190).

* symtab.cc (symbol_table::fcn_info::fcn_info_rep::load_class_constructor):
Add constructor to list of class_methods.

diff --git a/libinterp/corefcn/symtab.cc b/libinterp/corefcn/symtab.cc
--- a/libinterp/corefcn/symtab.cc
+++ b/libinterp/corefcn/symtab.cc
@@ -429,16 +429,17 @@ symbol_table::fcn_info::fcn_info_rep::lo
       octave_function *fcn = load_fcn_from_file (file_name, dir_name, name,
                                                  package_name);
 
       if (fcn)
         {
           retval = octave_value (fcn);
 
           class_constructors[name] = retval;
+          class_methods[name] = retval;
         }
     }
   else
     {
       // Classdef constructors can be defined anywhere in the path, not
       // necessarily in @-folders. Look for a normal function and load it.
       // If the loaded function is a classdef constructor, store it as such
       // and restore function_on_path to its previous value.
@@ -451,16 +452,17 @@ symbol_table::fcn_info::fcn_info_rep::lo
         {
           octave_function *fcn = maybe_cdef_ctor.function_value (true);
 
           if (fcn && fcn->is_classdef_constructor ())
             {
               retval = maybe_cdef_ctor;
 
               class_constructors[name] = retval;
+              class_methods[name] = retval;
 
               function_on_path = old_function_on_path;
             }
         }
     }
 
   return retval;
 }
